src/server/generic_family.cc

```diff
@@ -587,6 +587,15 @@ OpStatus OpExpire(const OpArgs& op_args, string_view key, const DbSlice::ExpireP
   if (!IsValid(it))
     return OpStatus::KEY_NOTFOUND;
 
+  // Convert all kinds to PEXPIREAT for replication, keep nx options.
+  if (op_args.tx->HasJournal()) {
+    char cmd[] = "PEXPIREAT";
+    if (auto time = params.Calculate(op_args.db_cntx.time_now_ms); time.ok()) {
+      auto time_str = absl::StrCat(time.value().second);
+      op_args.tx->LogJournal({{0, MutableSlice{cmd}}, {1, MutableSlice{time_str}}});
+    }
+  }
+
```

Comment:
I did this to keep the arguments, but we can just recover them from the expire params and issue a fully fresh transaction